### PR TITLE
Pavan/added trip functions

### PR DESCRIPTION
**Pull request by pavan3008**
# Proposed Changes

Added trip related functions along with the respective UI.

|Screenshots|
|---|
|Simulator Screenshot - iPhone 14 Pro - 2023-06-03 at 14 56 07|
|Simulator Screenshot - iPhone 14 Pro - 2023-06-03 at 14 54 01|
|Simulator Screenshot - iPhone 14 Pro - 2023-06-03 at 14 59 27|

# Checklist

- [x]  This has been tested
- [x]  Self-documenting code
- [x]  Code is linted

# Instructions to review

Please review the code.

### CODE REVIEW

File: SmilesMiles/Model/TripModel.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
+//
+//  TripModel.swift
+//  SmilesMiles
+//
+//  Created by Pavan Sai Nallagoni on 5/27/23.
+//
+
+struct Trip: Codable, Equatable {
+    let tripID: String
```
Comment (KedarDhere) [inline SmilesMiles/Model/TripModel.swift]: Follow the same convention everywhere while writing ID. Use either ID or Id

File: SmilesMiles/Model/TripModel.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    let tripName: String
+    let status: String
```
Comment (KedarDhere) [inline SmilesMiles/Model/TripModel.swift]: Check if status needs to be var, as status might change from In progress to Complete

File: SmilesMiles/View/TripView.swift [hunk 3]
```diff
@@ -15,64 +16,84 @@ struct TripView: View {
...
-                        Text(user.name)
-                        Text(user.email)
-                        // Display other user details as needed
-                    }
-                    
                     List {
-                        ForEach(viewModel.filteredTrips(), id: \.self) { trip in
-                            NavigationLink(destination: TripDetail(tripName: trip, numberOfTrips: $viewModel.numberOfTrips, tripViewModel: viewModel)) {
-                                Text(trip)
+                        ForEach(viewModel.filteredTrips(), id: \.tripID) { trip in
```
Comment (KedarDhere) [inline SmilesMiles/View/TripView.swift]: Check if ForEach is required for navigationLink

**Review by KedarDhere — APPROVED** [review]
Looks good to me

File: SmilesMiles/View/TripView.swift (hunk 3 above)
Comment (pavan3008) [reply SmilesMiles/View/TripView.swift]: Thank you for bringing it to my attention. I have learned that the List view does not have a direct onDelete modifier, so I need to utilize ForEach instead.

`Value of type 'List<Never, ForEach<[Trip], String, NavigationLink<VStack<TupleView<(Text, some View)>>, TripDetail>>>' has no member 'onDelete'`

**Comment by pavan3008** [comment]
Thanks for taking time to review the code 💯

**Pull request merged**